lte/gateway/fabfile.py

```diff
@@ -59,147 +65,154 @@
+                    mirrored_packages_file += '_focal'
+                if vm and vm.startswith('magma_'):
+                    mirrored_packages_file += vm[5:]
+
+                c_agw.run(
+                    f'cat {mirrored_packages_file}'
+                    + ' | xargs -I% sudo aptitude download -q2 %',
+                )
+                c_agw.run('cp *.deb ~/magma-packages')
+                c_agw.run('sudo rm -f *.deb')
+
+            if all_deps:
+                pkg.download_all_pkgs(c_agw)
+                c_agw.run('cp /var/cache/apt/archives/*.deb ~/magma-packages')
 
-        # Copy out C executables into magma-packages as well
-        _copy_out_c_execs_in_magma_vm()
+            # Copy out C executables into magma-packages as well
+            _copy_out_c_execs_in_magma_vm(c_agw)
 
 
-def openvswitch(destroy_vm='False', destdir='~/magma-packages/'):
-    destroy_vm = strtobool(destroy_vm)
+@task
+def openvswitch(c, destroy_vm=False, destdir='~/magma-packages/'):
     # If a host list isn't specified, default to the magma vagrant vm
-    if not env.hosts:
-        vagrant_setup('magma', destroy_vm=destroy_vm)
-    run('~/magma/third_party/gtp_ovs/ovs-gtp-patches/2.15/build.sh ' + destdir)
+    c_agw = vagrant_connection(c, 'magma', destroy_vm=destroy_vm)
+    with c_agw:
+        c_agw.run('~/magma/third_party/gtp_ovs/ovs-gtp-patches/2.15/build.sh ' + destdir)
 
 
-def depclean():
+@task
+def depclean(c):
     '''Remove all generated packaged for dependencies'''
     # If a host list isn't specified, default to the magma vagrant vm
-    if not env.hosts:
-        setup_env_vagrant()
-    run('rm -rf ~/magma-deps')
+    c_agw = vagrant_connection(c, 'magma')
+    with c_agw:
+        c_agw.run('rm -rf ~/magma-deps')
 
 
-def upload_to_aws():
+@task
+def upload_to_aws(c):
     # If a host list isn't specified, default to the magma vagrant vm
-    if not env.hosts:
-        setup_env_vagrant()
+    c_agw = vagrant_connection(c, 'magma')
+    with c_agw:
+        pkg.upload_pkgs_to_aws(c_agw)
 
-    pkg.upload_pkgs_to_aws()
 
+@task
+def copy_packages(c):
+    c_agw = vagrant_connection(c, 'magma')
+    with c_agw:
+        pkg.copy_packages(c_agw)
 
-def copy_packages():
-    if not env.hosts:
-        setup_env_vagrant()
-    pkg.copy_packages()
 
-
-def s1ap_setup_cloud():
+@task
+def s1ap_setup_cloud(c):
     """ Prepare VMs for s1ap tests touching the cloud. """
     # Use the local cloud for integ tests
-    setup_env_vagrant()
-    connect_gateway_to_cloud(None, DEFAULT_CERT)
+    c_agw = vagrant_connection(c, "magma")
+    with c_agw:
+        connect_gateway_to_cloud(c_agw, None, DEFAULT_CERT)
 
-    # Update the gateway's streamer timeout and restart services
-    run("sudo mkdir -p /var/opt/magma/configs")
-    _set_service_config_var('streamer', 'reconnect_sec', 3)
+        # Update the gateway's streamer timeout and restart services
+        c_agw.run("sudo mkdir -p /var/opt/magma/configs")
```

Comment:
Could we use `c_agw.sudo(` here and in the lines below or would that behave differently? If it is the same then do a full-text search for `c_agw.run("sudo`, because there are a couple other places as well.

Reply:
It should be possible here. As stated in another comment above, `c.sudo` does not work within a `c.cd` context, but should be fine otherwise (which brings the question if we should just switch to `c.run('sudo ....)` in general)